client/src/includes/sidePanel.js

```diff
@@ -99,6 +102,123 @@ export default function initSidePanel() {
     });
   }
 
+  const resizeGrip = document.querySelector('[data-form-side-resize-grip]');
+  const widthInput = document.querySelector('[data-form-side-width-input]');
+  const blankCanvas = document.createElement('canvas');
+
+  const getSidePanelWidthStyles = () => {
+    const sidePanelStyles = getComputedStyle(sidePanelWrapper);
+    const minWidth = parseFloat(sidePanelStyles.minWidth);
+    const maxWidth = parseFloat(sidePanelStyles.maxWidth);
+    const width = parseFloat(sidePanelStyles.width);
+    const range = maxWidth - minWidth;
+
+    return { minWidth, maxWidth, width, range };
+  };
+
+  const setSidePanelWidth = (width) => {
+    const { minWidth, range, width: currentWidth } = getSidePanelWidthStyles();
+    const newWidth = parseFloat(width)
+      ? Math.max(minWidth, Math.min(width, minWidth + range))
+      : currentWidth;
+
+    sidePanelWrapper.style.width = `${newWidth}px`;
+    widthInput.value = ((newWidth - minWidth) / range) * 100;
+    localStorage.setItem('wagtail:side-panel-width', newWidth);
+  };
+
+  resizeGrip.addEventListener('dragstart', (e) => {
+    // When dragging, replace 'ghost' image of the element with a blank canvas.
+    // It needs to be in the DOM, so add it temporarily and remove on dragend.
+    document.body.appendChild(blankCanvas);
+    e.dataTransfer.effectAllowed = 'move';
+    e.dataTransfer.setDragImage(blankCanvas, 0, 0);
+
+    document.body.classList.add('side-panel-resizing');
+  });
+
+  resizeGrip.addEventListener('dragend', () => {
+    document.body.classList.remove('side-panel-resizing');
+    document.body.removeChild(blankCanvas);
+    resizeGrip.focus();
+  });
+
+  resizeGrip.addEventListener('drag', (e) => {
+    // If the grip is dropped on certain elements,
+    // sometimes the event fires with screenX of 0, so ignore it.
+    if (!e.screenX) return;
+
+    const { width } = getSidePanelWidthStyles();
+    const newWidth = width - e.offsetX;
+    setSidePanelWidth(newWidth);
+  });
+
+  // Prevent any default dragover behaviour on the document.
+  document.addEventListener('dragover', (e) => e.preventDefault());
```

Comment:
Again this comment just describes the code rather than explain why we do this. Could you explain the "why" instead?

Is there any way we could only prevent those events if they come from the resize grip in particular? Doing this for all dragover events feels like it could have unintended consequences.

Reply:
This is no longer needed in the new implementation.